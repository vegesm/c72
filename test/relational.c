main() {
    extern iarr[];
    int a, b;
    char c ,d, carr[4];

    if(2>1) {
        printf("test1 - ok\n");
    }

    a = 3;
    if(a>1) {
        printf("test2 - ok\n");
    }

    a = 3;
    b = -4;
    if(a>b) {
        printf("test3 - ok\n");
    }

    if(iarr[1]>1) {
        printf("test4 - ok\n");
    }

    if(iarr[1]>2+2) {
        printf("test5 - ok\n");
    }

    c = 6;
    d = -7;
    if(c>1) {
        printf("test6 - ok\n");
    }

    if(c>d) {
        printf("test7 - ok\n");
    }

    carr[0] = 2;
    carr[1] = -128;
    if(d>carr[1]) {
        printf("test8 - ok\n");
    }

    printf("eval - rel1: %d\n", a>=3);
    printf("eval - rel2: %d\n", a>=9);
}


iarr[] 1, 8, 10, 128;

