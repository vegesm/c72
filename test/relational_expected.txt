test1 - ok
test2 - ok
test3 - ok
test4 - ok
test5 - ok
test6 - ok
test7 - ok
test8 - ok
eval - rel1: 1
eval - rel2: 0
